dbms/src/Interpreters/JoinHashMap.h

```diff
@@ -33,15 +33,29 @@ struct RowRef
     {}
 };
 
+struct CachedColumnInfo
+{
+    std::mutex mu;
+    Columns columns;
+    bool generate_cached_columns = false;
```

Comment:
An enum value instead of boolean value may make it more clean.

Reply:
OK, add an enum here.